dataflow/src/main/java/org/checkerframework/dataflow/analysis/ConditionEvaluator.java

```diff
@@ -0,0 +1,110 @@
+package org.checkerframework.dataflow.analysis;
+
+import org.checkerframework.dataflow.cfg.node.ConditionalAndNode;
+import org.checkerframework.dataflow.cfg.node.ConditionalNotNode;
+import org.checkerframework.dataflow.cfg.node.ConditionalOrNode;
+import org.checkerframework.dataflow.cfg.node.EqualToNode;
+import org.checkerframework.dataflow.cfg.node.GreaterThanNode;
+import org.checkerframework.dataflow.cfg.node.GreaterThanOrEqualNode;
+import org.checkerframework.dataflow.cfg.node.LessThanNode;
+import org.checkerframework.dataflow.cfg.node.LessThanOrEqualNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.node.NotEqualNode;
+
+public class ConditionEvaluator<A extends AbstractValue<A>, S extends Store<S>> {
+
+    public enum ConditionalFlow {
+        THEN,
+        ELSE,
+        BOTH
+    }
+
+    public ConditionalFlow visit(Node node, TransferInput<A, S> in) {
+        if (node instanceof GreaterThanNode) {
```

Comment:
Shouldn't this class implement https://github.com/typetools/checker-framework/blob/master/dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/NodeVisitor.java  or build on https://github.com/typetools/checker-framework/blob/master/dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/AbstractNodeVisitor.java?